File: android-client/data/src/main/java/ru/hits/hubabank/data/network/credit/model/CreditDto.kt
package ru.hits.hubabank.data.network.credit.model

import kotlinx.serialization.Serializable
import ru.hits.hubabank.domain.credit.model.Credit
import java.time.LocalDate

@Serializable
internal data class CreditDto(
    val id: String,
    val accountId: String,
    val billId: String,
    val completionDate: String,
    val lastArrearsUpdate: String,
    val interestRate: Float,
    val collectionDay: Int,
    val principal: Long,
    val currentAccountsPayable: Long,
    val arrearsInterest: Long,
    val arrears: Long,
    val fine: Long,
)

internal fun CreditDto.toDomain(): Credit {
    return Credit(
        id = id,
        accountId = accountId,
        billId = billId,
        completionDate = LocalDate.parse(completionDate),
        lastArrearsUpdate = LocalDate.parse(lastArrearsUpdate),
        interestRate = interestRate,
        collectionDay = collectionDay,
        principal = principal,
        currentAccountsPayable = currentAccountsPayable,
        arrearsInterest = arrearsInterest,
        arrears = arrears,
        fine = fine,
    )
}

File: BFF-client/BFF-client.Api/Models/Credit/CreditDto.cs
namespace BFF_client.Api.Models.Credit
{
    public class CreditDto
    {
        public Guid Id { get; set; }

        public Guid AccountId { get; set; }

        public Guid BillId { get; set; }

        public DateOnly CompletionDate { get; set; }

        public DateOnly LastArrearsUpdate { get; set; }

        public float InterestRate { get; set; }

        public int CollectionDay { get; set; }

        public long Principal {  get; set; }

        public long CurrentAccountsPayable {  get; set; }

        public long ArrearsInterest {  get; set; }

        public long Arrears {  get; set; }

        public long Fine {  get; set; }
    }
}
